multimodal/src/autogluon/multimodal/models/sam.py

```diff
@@ -2,20 +2,278 @@
 import os
 import time
 import warnings
-from typing import Optional
+from typing import Dict, List, Optional, Tuple
 
 import torch
 import torch.nn.functional as F
 from omegaconf import DictConfig
 from torch import nn
 from transformers import SamConfig, SamModel
+from transformers.models.sam.modeling_sam import SamImageSegmentationOutput
 
-from ..constants import COLUMN, IMAGE, IMAGE_VALID_NUM, LABEL, LOGITS, SEMANTIC_SEGMENTATION
+from ..constants import (
+    CLASS_LABEL,
+    CLASS_LOGITS,
+    COLUMN,
+    IMAGE,
+    IMAGE_VALID_NUM,
+    LABEL,
+    LOGITS,
+    MASK_LABEL,
+    MASK_SEMANTIC_INFER,
+)
 from .utils import assign_layer_ids, freeze_model_layers
 
 logger = logging.getLogger(__name__)
 
 
+def multi_class_mask_decoder_forward(
```

Comment:
Inside the function, better to mark which are original code and which are new added.

Reply:
Add the original link of the copied code.